Flaskerizer_src/gui_for_flaskerizer.py

```diff
@@ -0,0 +1,185 @@
+
+
+        self.label_js = tk.Label(self.main,
+                                 text="JavaScript File Location:")
+        self.label_js.grid(row=2, column=0, sticky="WENS", pady=20)
+        self.folder_search_js = tk.Button(self.main,
+                                          text="Search Files", command=self.get_js_folder)
+        self.folder_search_js.grid(row=2, column=1, sticky="WENS", pady=20)
+        self.folder_location_js = tk.Entry(self.main,
+                                           textvariable=self.js_location, width=80)
+        self.folder_location_js.grid(row=2, column=2, sticky="WENS", pady=20)
+
+
+        self.main.pack()
+        self.ok_button = tk.Button(self.root,
+                                   text="OK", command=self.get_values).pack()
+
+    def get_html_folder(self):
+        """Opens a file dialog prompting the user to select an HTML file.
+        Gets the directory path with the path_to_folder function
+        and sets the html_location variable with the returned path value.
+        """
+
+        path = filedialog.askopenfilename(title= "Select the main html file normally index.html",
+        filetypes=(("Html Files", "*.html"), ("all files", "*.*")))
+        self.html_location.set(path)
+        if self.validate_path(path):
+            self.html_location.set(path)
+        else:
+            self.html_location.set("Please select a valid html file")
+
+    def get_static_folder(self):
+        """Opens a file dialog prompting the user to select the static folder.
+        Gets the path, and sets the static_location to the path value.
+        """
+
+        path = filedialog.askdirectory(title="Select the main static folder")
+        if self.validate_path(path):
+            self.static_location.set(path)
+        else:
+            self.static_location.set("Please select a valid static folder")
+
+    def get_js_folder(self):
+        """Opens a file dialog prompting the user to select a JavaScript file.
+        Gets the directory path with the path_to_folder function
+        and sets the js_location variable with the returned path value.
+        """
+
+        path = filedialog.askopenfilename(title = "Select a .js file, normally inside the js folder",
+        filetypes=(("JavaScript Files", "*.js"), ("all files", "*.*")))
+        if self.validate_path(path):
+            self.js_location.set(path)
+        else:
+            self.js_location.set("Please select a valid js file")
+
+    def path_to_folder(self, path):
+        """Transforms a File path into a folder path
+        stripping the file name from the path,
+        leaving only the directory path.
+        """
+
+        return os.path.split(path)[0]
+    def validate_path(self, path):
+        """Validates the path making sure it exists
+        """
+        if os.path.exists(path):
+            return True
+        else:
+            return False
+
+    def validate_entries(self):
+        """Validates all entries and if
```

Comment:
Maybe you can comment back and let me know, but isn't there validation of entries as the user is selecting the paths? I am trying to imagine a scenario of how the user can select invalid entries in the first place and have this function return false.

Reply:
I added it because manual input is an option as well since they are entries the user can type a path or copy & paste on it.